packages/medusa/src/api/routes/store/collections/index.ts

```diff
@@ -1,21 +1,38 @@
 import { Router } from "express"
-import { PaginatedResponse } from "./../../../../types/common"
+import { PaginatedResponse } from "../../../../types/common"
 import { ProductCollection } from "../../../../"
-import middlewares from "../../../middlewares"
+import middlewares, { transformQuery } from "../../../middlewares"
+import { StoreGetCollectionsParams } from "./list-collections"
 
 const route = Router()
 
 export default (app) => {
   app.use("/collections", route)
 
-  route.get("/", middlewares.wrap(require("./list-collections").default))
+  route.get(
+    "/",
+    transformQuery(StoreGetCollectionsParams, {
+      allowedFields,
+      isList: true,
+    }),
+    middlewares.wrap(require("./list-collections").default)
+  )
   route.get("/:id", middlewares.wrap(require("./get-collection").default))
 
   return app
 }
 
-export const defaultStoreCollectionFields = ["id", "title", "handle"]
 export const defaultStoreCollectionRelations = ["products"]
+export const allowedFields = [
+  "id",
+  "title",
+  "handle",
+  "metadata",
+  "created_at",
+  "updated_at",
+  "deleted_at",
+  ...defaultStoreCollectionRelations,
```

Comment:
question: Is this intentionally included? `products` is a relation, not a field

Reply:
yes it is intentionally  since this collection of string is the one compared to what is provided for both fields and relations.